platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/MerkleStateRoot.java

```diff
@@ -1014,20 +1014,28 @@ public PlatformStateAccessor getReadablePlatformState() {
      */
     @NonNull
     @Override
-    public PlatformStateAccessor getWritablePlatformState() {
+    public PlatformStateModifier getWritablePlatformState() {
         if (isImmutable()) {
             throw new IllegalStateException("Cannot get writable platform state when state is immutable");
         }
         return writablePlatformStateStore();
     }
 
     /**
-     * Updates the platform state with the values from the provided instance of {@link PlatformStateAccessor}
+     * {@inheritDoc}
+     */
+    @Override
+    public void initPlatformState() {
+        writablePlatformStateStore();
```

Comment:
Perhaps it would be worth extracting the relevant logic from `writablePlatformStateStore`. Looking at that method, it appears there is some stuff in the current `writablePlatformStateStore` method which doesn't actually need to be done on init

Reply:
I replaced it with more specific call